src/app_service/service/contacts/service.nim

```diff
@@ -395,9 +392,9 @@ QtObject:
     if self.contacts.hasKey(publicKey):
       if self.contacts[publicKey].dto.added and not self.contacts[publicKey].dto.removed and contact.added and not contact.removed:
         signal = SIGNAL_CONTACT_UPDATED
-    if contact.removed:
-      singletonInstance.globalEvents.showContactRemoved("Contact removed", fmt "You removed {contact.displayName} as a contact", contact.id)
-      signal = SIGNAL_CONTACT_REMOVED
+      if contact.removed and not self.contacts[publicKey].dto.removed:
+        singletonInstance.globalEvents.showContactRemoved("Contact removed", fmt "You removed {contact.displayName} as a contact", contact.id)
```

Comment:
I don't think we need to do the notification here at all; we do those (in the correct format) in Popups.qml already. There are different permutations, depending on what other actions the user takes via the popups (like marking as untrosted, blocking, etc). All of these are processed at once, so that you don't get a single notification for each action.

Reply:
Also (not a big deal atm) this is not translatable at all

Reply:
That's true. I guess we have a couple of stray notifications in Nim. I don't know if we have already an issue to refactor those to move them to QML

Reply:
Yeah, there was some initial effort by Noelia (see ToastsManager.qml) but never finished actually; I guess here's one of those (many) still done on the NIM side